components/builder-api/src/server/error.rs

```diff
@@ -191,7 +191,7 @@ impl Into<HttpResponse> for Error {
             Error::BadRequest(_) => HttpResponse::new(StatusCode::BAD_REQUEST),
             Error::Github(_) => HttpResponse::new(StatusCode::FORBIDDEN),
             Error::CircularDependency(_) => HttpResponse::new(StatusCode::FAILED_DEPENDENCY),
-            Error::NetError(ref e) => HttpResponse::build(net_err_to_http(&e)).json(&e),
+            Error::NetError(ref e) => HttpResponse::new(net_err_to_http(&e)),
```

Comment:
I think @cnunciato needed this to get the polling on log watch working. A 404 wasn't enough to tell him that the request was dead but to keep watching so he was using the `{"code": 4}` in the response body

Reply:
Right, so there's more more 404 for missing job logs.  The only remaining case is if there's a legit failure.  That said, @cnunciato may have another commit to add to this.